### DISCUSSION TITLE
pnpm integrity is sha1 on verdaccio proxy and sha512 on npmjs

### DISCUSSION THREAD

**Discussion by viceice** [verdaccio/verdaccio#4550] [Q&A]
Hi.

I'm using normal npmjs registry to fetch public packages when outside of our company (to not kill the vpn).
When inside the company we set `npm_config_registry=https://verdaccio.domain.test/` via env to use verdaccio pull through proxy.

When i update packages throug verdaccio i only get sha1 integrity hashes, when I'm outside i get the expected sha512 hashes.

Is this expected behavior or a bug?

It would be nice to support sha512 through verdaccio too.

**Comment by juanpicado** [comment]
Yeah that's something requires some update I also noticed while ago.  I haven't research much in that discussion so any help here is very welcome.

**Reply by viceice** [reply]
ok, will try to do some research the next days

**Comment by viceice** [comment]
Was there any recent change? I now see sha512 hashes as expected 🤔 

![image](https://github.com/verdaccio/verdaccio/assets/1798109/99faad19-fbd2-40e4-8fde-c6224ed2b830)

![image](https://github.com/verdaccio/verdaccio/assets/1798109/6523ebeb-ab3e-4616-9271-3d45de8a5101)

**Comment by juanpicado** [comment]
> Was there any recent change? I now see sha512 hashes as expected 🤔
> 
> ![image](https://private-user-images.githubusercontent.com/1798109/320465277-99faad19-fbd2-40e4-8fde-c6224ed2b830.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODAyOTYsIm5iZiI6MTcxMjU3OTk5NiwicGF0aCI6Ii8xNzk4MTA5LzMyMDQ2NTI3Ny05OWZhYWQxOS1mYmQyLTQwZTQtOGZkZS1jNjIyNGVkMmI4MzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDhUMTIzOTU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NGEwYzkwNTk3OTQ1MzQwNzU2Y2Q5NjRhNmRkYTAxMWI0OTVjNzU0NTMyOTZiNTkyODAzNTY0OTI3MzBjYzQ3MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.PRyaQb944gLc5xMFDkID3-i3SlSqlbME88MEqt-L0IE)
> 
> ![image](https://private-user-images.githubusercontent.com/1798109/320465488-6523ebeb-ab3e-4616-9271-3d45de8a5101.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI1ODAyOTYsIm5iZiI6MTcxMjU3OTk5NiwicGF0aCI6Ii8xNzk4MTA5LzMyMDQ2NTQ4OC02NTIzZWJlYi1hYjNlLTQ2MTYtOTI3MS0zZDQ1ZGU4YTUxMDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDQwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA0MDhUMTIzOTU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDA0YzMwNDhhYWYwMjhkMGU0MjdmMzg5ZmNiZTg2YmI3ZmMyMWQwODhlMGZjNjViZGVjYmEyYzUxNzI0MzkwZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.u9wWpjKVFY3oPxkOP17N9_djg-LG0kNzTIzur5E1bDU)

Nop changes on storage.
